sklearn/manifold/locally_linear.py

```diff
@@ -68,10 +68,9 @@ def barycenter_kneighbors_graph(X, n_neighbors, reg=1e-3, n_jobs=1):
 
     Parameters
     ----------
-    X : {array-like, sparse matrix, BallTree, KDTree, NearestNeighbors}
+    X : array-like
```

Comment:
I tried passing in ball trees and kd trees directly to the function and failed on the call to

     data = barycenter_weights(X, X[ind], reg=reg)

due to the indexing. If NearestNeighbors et al. all actually work I can certainly restore that.